### PR TITLE
Reload weights after plateau

### PR DESCRIPTION
**Pull request by DanBmh**
Reload checkpoint weights after reaching a plateau that we use the best_dev weights again

### CODE REVIEW

**Comment by lissyx** [comment]
@DanBmh Thanks, can you elaborate a little bit? My mind is kind of somewhere else, so I'm unsure I get the point here.

**Comment by DanBmh** [comment]
Currently training looks like this:
```
epoch 5: val_loss=62
e6: vl=59
e7: vl=60
e8: vl=61
Reached a plateau, LearningRate:=LR*0.1
e9: vl=60       <- Here we're using the weights from e8, with the suggested changes we're using e6 instead
e10: vl=58   <- We have an improvement but the network has to do some more work to fix the errors from e7+e8
```

The old approach did still work well but I think we can make it even better by reloading the weights from the best_dev checkpoint.

**Comment by DanBmh** [comment]
Not ready yet! 
Found an error when using `--drop_source_layers` flag

**Comment by DanBmh** [comment]
Working again:)

File: training/mozilla_voice_stt_training/train.py [hunk 1]
```diff
@@ -645,6 +645,9 @@ def __call__(self, progress, data, **kwargs):
                         log_info('Encountered a plateau, reducing learning rate to {}'.format(
                             current_learning_rate))
 
+                        # Reload checkpoint that we use the best_dev weights again
+                        load_or_init_graph_for_training(session, allow_drop_layers=False)
```
Comment (reuben) [inline training/mozilla_voice_stt_training/train.py:649]: Please keep this function unchanged and add a new explicitly `load_best_checkpoint` function, we don't want this call to load last silently for example.

**Review by reuben — CHANGES_REQUESTED** [review]

**Review by reuben — APPROVED** [review]

**Comment by reuben** [comment]
Re-opened as #3261 to run tests.

**Comment by reuben** [comment]
Merged in #3261

**Pull request closed without merging**